addons/mail/static/src/components/attachment/attachment_tests.js

```diff
@@ -667,6 +667,98 @@ QUnit.test('[technical] does not crash when the viewer is closed before image lo
     }
 });
 
+QUnit.test('text attachment is viewable', async function (assert) {
+    assert.expect(1);
+
+    await this.start();
+    const attachment = this.env.models['mail.attachment'].create({
+        filename: "test.txt",
+        id: 750,
+        mimetype: 'text/plain',
+        name: "test.txt",
+    });
```

Comment:
Any way to use `this.data` instead? Maybe a message test (or even thread view test if must be) that uses mock server and test an actual flux displaying such attachment. Or keep an attachment test but make the fetch RPC manually?

Maybe check with @alexkuhn for preferred solution

Reply:
Not easy to use this.data as you'll need to show the GUI of an element accepting uploads and showing them (like a message lit and its composer) and you'll need to play with handling uploading events and so on ... All the test file is not using the this.data method, so I think we should go this way and plan a refactoring of all the file which is, I think, not that easy

Reply:
It's ok for me if we don't change this in this PR.